lib/install/management/create.go

```diff
@@ -100,6 +106,67 @@ func (d *Dispatcher) createPool(conf *config.VirtualContainerHostConfigSpec, set
 	return nil
 }
 
+func (d *Dispatcher) createVMGroup(conf *config.VirtualContainerHostConfigSpec) error {
+	defer trace.End(trace.Begin("", d.op))
+
+	if !conf.UseVMGroup {
+		return nil
+	}
+
+	// TODO[zjs]: need to check vmgroup name constraints vs resource pool constraints
+	vmGroupName := conf.Name
+	d.op.Debugf("Creating DRS VM Group %q on %q", vmGroupName, d.appliance.Cluster)
+
+	spec := &types.ClusterConfigSpecEx{
+		GroupSpec: []types.ClusterGroupSpec{
+			{
+				ArrayUpdateSpec: types.ArrayUpdateSpec{
+					Operation: types.ArrayUpdateOperationAdd,
+				},
+				Info: &types.ClusterVmGroup{
+					ClusterGroupInfo: types.ClusterGroupInfo{
+						Name: vmGroupName,
+					},
+					Vm: []types.ManagedObjectReference{d.appliance.Reference()},
+				},
+			},
+		},
+	}
+
+	_, err := tasks.WaitForResult(d.op, func(op context.Context) (tasks.Task, error) {
+		d.op.Debugf("Attempting to create VM group %q for VCH.", vmGroupName)
+		t, err := d.appliance.Cluster.Reconfigure(op, spec, true)
+
+		// Unfortunately, WaitForResult treats InvalidArgument as retry-able. In this case, it is not. So let's get the
+		// error and if it's an InvalidArgument, we'll replaces it with a new error. This is terrible for a variety of
+		// reasons, but it's unclear just how much code may be relying on WaitForResult's current behavior to handle
+		// potential race-like cases (see, e.g., discussion in #4597).
```

Comment:
My preference for starting to untangle this would be:
a. have the bug fixed and stop supporting versions without it (not viable)
a'. identify which paths are susceptible to this vSAN bug
b. have a separate function that wraps the _actual_ set of transient errors and adds InvalidArguement, specifically checking for InvalidPath and vSAN.
c. use that extended function on paths that are prone to it.

a' does not need to happen in this PR but the below code may be neater if we have a second method called `isTransientError` that does not include the vSAN issue that we can use in most placed. I mention this because @AngieCris needs a retry that doesn't include InvalidArgument, but instead adds in NoSuchManagedObject for concurrent listing/iteration.

Reply:
Yeah, I don't think the work-around in this PR is the Right Thing, but for now I'm focusing on getting things working reliably.

Reply:
FYI as part of PR https://github.com/vmware/vic/pull/7617 I included `isTransientError` as @hickeng suggested above and make `isRetryError` just a wrapper around it. You should be able to directly use `isTransientError` later after the PR is in

Reply:
> We should be able to directly use `isTransientError` later if we don't want to retry on `InvalidArgument` fault, after the PR is in

Will do!